sdk/identity/Azure.Identity/src/KubernetesProxyHttpHandler.cs

```diff
@@ -0,0 +1,229 @@
+
+using System;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+
+namespace Azure.Identity
+{
+    /// <summary>
+    /// HTTP message handler that redirects all requests to a Kubernetes token proxy.
+    /// Handles custom CA certificates and monitors CA file changes for automatic reloading.
+    /// </summary>
+    internal class KubernetesProxyHttpHandler : DelegatingHandler
+    {
+        private readonly KubernetesProxyConfig _config;
+        private HttpClientHandler _innerHandler;
+        private byte[] _lastCaData;
+        private readonly object _lock = new object();
+        private SemaphoreSlim _requestSemaphore;
+        private int _requestCount;
+
+        public KubernetesProxyHttpHandler(KubernetesProxyConfig config)
+        {
+            _config = config ?? throw new ArgumentNullException(nameof(config));
+
+            _innerHandler = CreateHandler();
+            InnerHandler = _innerHandler;
+            _requestSemaphore = new SemaphoreSlim(1, 1);
+            _requestCount = 0;
+        }
+
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            Interlocked.Increment(ref _requestCount);
+
+            try
+            {
+                // Reload handler if CA certificate file has changed
+                if (ShouldReloadHandler())
+                {
+                    lock (_lock)
+                    {
+                        if (ShouldReloadHandler())
+                        {
+                            var oldHandler = _innerHandler;
+                            var oldSemaphore = _requestSemaphore;
+
+                            _innerHandler = CreateHandler();
+                            InnerHandler = _innerHandler;
+                            _requestSemaphore = new SemaphoreSlim(1, 1);
+
+                            // Dispose old handler after in-flight requests complete
+                            Task.Run(async () =>
+                            {
+                                await oldSemaphore.WaitAsync().ConfigureAwait(false);
+                                oldHandler?.Dispose();
+                                oldSemaphore?.Dispose();
+                            });
+                        }
+                    }
+                }
+
+                RewriteRequestUrl(request);
+
+                return await base.SendAsync(request, cancellationToken).ConfigureAwait(false);
+            }
+            finally
+            {
+                if (Interlocked.Decrement(ref _requestCount) == 0)
+                {
+                    _requestSemaphore?.Release();
+                }
+            }
+        }
+
+        private void RewriteRequestUrl(HttpRequestMessage request)
+        {
+            if (request.RequestUri == null)
+            {
+                return;
```

Comment:
I guess this condition is just satisfying the compiler and will never happen in practice, but in principle shouldn't we throw when we get a request with no URI?